### DISCUSSION TITLE
Media without dates in EXIF metadata have "Unknown" date

### DISCUSSION THREAD

**Discussion by uqlel** [photoprism/photoprism#3598] [Q&A]
#### 1. What is not working as documented?

Media files without date in the metadata [should be used modification date as fallback](https://github.com/photoprism/photoprism/issues/3575#issuecomment-1650067689), but they are just using "Unknown" 

#### 2. How can we reproduce it?

1. Install PhotoPrism via openmediavault on Raspberry Pi 4
2. Enable PhotoPrism with the originals, app data and import folders on an external HDD USB drive.
3. Import all files from the import folder.
4. Enter "Search" tab.
5. Some files (mostly from social medias that probably do not provide EXIF metadata) are saying that the date is unknown. 

#### 3. What behavior do you expect?

I except them to show the right date, because they are sorted in the right order in the "Originals" folder.

#### 4. What could be the cause of your problem?

It could be no support for searching files without date in the metadata, 

#### 5. Can you provide us with example files for testing, error logs, or screenshots?
Example video, that is showing "Unknown" date in the UI, but has a filename with the right date in the Originals folder.
![image](https://github.com/photoprism/photoprism/assets/59146379/088590dc-9420-463d-9ca4-16ad2f529b39)
![image](https://github.com/photoprism/photoprism/assets/59146379/452057ec-be1c-48e4-9845-0496bcc36eb0)
(the "Time UTC" is probably the import time, because I left my Raspberry Pi on for the night to import my photos)
![image](https://github.com/photoprism/photoprism/assets/59146379/2d2e7231-4dc6-4e31-9b00-2df37ea171f1)

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: ARM64
(b) Database Type & Version: I don't really know, OpenMediaVault manages it for me
(c) Operating System Types & Versions: Raspberry Pi OS 64bit with OpenMediaVault installed, it manages PhotoPrism
(d) Browser Types & Versions: Latest LibreWolf, Latest Google Chrome
(e) Ad Blockers, Browser Plugins, and/or Firewall Software: uBlock and Privacy Badger, but it worked the same 

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: Raspberry Pi 4
(b) Physical Memory & Swap Space in GB: 4GB RAM, 100MB swap
(c) Storage Type: USB 2,5" HDD

**Comment by uqlel** [comment]
All the screenshots also have the "Unknown" date, but they are in the "Review" tab. They still have dates in filenames though.

**Reply by lastzero** [reply]
The file names are generated automatically during import. If no information is found in the metadata or the original names, the file system's mod time is used for this. However, this is usually not the actual date when a photo was taken (or a graphic was created by the original author), but only the time when you downloaded or copied it.

**Reply by uqlel** [reply]
My library is managed that way, that the last modification dates are actually the creation dates. Is it possible to force PhotoPrism to use these dates to sort files in the library?

**Reply by lastzero** [reply]
You could add a config option for this or change the corresponding source code, but it is easiest to manually change the `taken_src` column in the `photos` table, for example to `meta` or `name`, to increase the confidence. Note that there are additionally the columns `photo_year`, `photo_month` and `photo_days`, which you can also set with an update query. Re-indexing might otherwise update them after you have changed the `taken_src` value.

**Reply by lastzero** [reply]
We have updated our User Guide FAQ to explicitly cover this question: https://docs.photoprism.app/user-guide/faq/#metadata

**Reply by lastzero** [reply]
Alternatively, you can organize your library in a way that the date is included in the folder and/or file names, so that it can be automatically picked up from there.

**Reply by uqlel** [reply]
> We have updated our User Guide FAQ to explicitly cover this question: https://docs.photoprism.app/user-guide/faq/#metadata

So it is not possible to automatically use the modification date as the creation date (without modifying the source code)?

**Reply by lastzero** [reply]
You have the options outlined above. For example, running an update statement on the photos table can do this in a single operation. However, there is no config option available at this time as such a behavior wasn't requested before and also doesn't make sense for most users.
